nistats/regression.py

```diff
@@ -347,41 +354,19 @@ def __init__(self, results):
         # put this as a parameter of LikelihoodModel
         self.df_resid = self.df_total - self.df_model
 
+        self.wdesign = results.model.wdesign
+
     def logL(self, Y):
         """
         The maximized log-likelihood
         """
-        raise ValueError('can not use this method for simple results')
+        raise ValueError('minimize_memory should be set to False to make residuals or predictions.')
```

Comment:
I don't understand what these methods are doing except raising minimize _memory error

Reply:
It looks like the goal of these errors is to give the user a useful error if they try to get voxelwise output from the model and they did not set `minimize_memory` to false, instead of the presumably less helpful error of trying to access a method that does not exist for an object. Is this correct @Gilles86?

Reply:
If that is the case, perhaps it will be better to set `minimize_memory=False` and issue a warning, instead of a pipeline stopping error.
WDYT @bthirion ?

Reply:
at this point it is too late anyway: these class (SimpleRegressionResults) holds results obtained when minimize_memory=True. the error says if you wanted to be able to compute residuals you should have used minimized_memory=False and obtained RegressionResults (without the simple) instead. arguably it should be a NotImplementedError. I think this is very useful because I remember users getting confused by RegressionResults/SimpleRegressionResults before, and it would help if the message was more verbose

Reply:
is this logL used anywhere ? If not, we should drop it.

Reply:
Removing logL is beyond the purview of this PR. I can remove it after this one is merged.